ui/src/app/machines/components/MachineHeaderForms/ActionFormWrapper/TagForm/hooks.ts

```diff
@@ -0,0 +1,29 @@
+import { useFormikContext } from "formik";
+import { useSelector } from "react-redux";
+
+import type { TagFormValues } from "./types";
+
+import type { RootState } from "app/store/root/types";
+import tagSelectors from "app/store/tag/selectors";
+import type { Tag, TagMeta } from "app/store/tag/types";
+import { toFormikNumber } from "app/utils";
+
+/**
+ * Get the tag objects for the tag ids that have been selected in the form.
+ */
+export const useSelectedTags = (): Tag[] => {
+  const { values } = useFormikContext<TagFormValues>();
+  const selectedTags = useSelector((state: RootState) =>
+    tagSelectors.getByIDs(
+      state,
+      values.tags.reduce<Tag[TagMeta.PK][]>((tagList, id) => {
+        const idNumber = toFormikNumber(id);
+        if (idNumber) {
+          tagList.push(idNumber);
+        }
+        return tagList;
+      }, [])
```

Comment:
This could probably be split out into its own variable, seems a bit muddled in the useSelector hook

Reply:
Good idea, I've also added a comment about why this is necessary.